packages/flutter/lib/src/widgets/animated_switcher.dart

```diff
@@ -343,6 +349,7 @@ class _AnimatedSwitcherState extends State<AnimatedSwitcher> with TickerProvider
       assert(!_outgoingEntries.contains(_currentEntry));
       _outgoingEntries.add(_currentEntry);
       _currentEntry.controller.reverse();
+      _updateTransitionForEntry(_currentEntry);
```

Comment:
We rebuild the child since its animation status changed from completed to reverse